### PR TITLE
[Localization] Localize pokemon form change message and translate

### PR DESCRIPTION
**Pull request by EnochG1**
## What are the changes?
Add the option to localize message text when pokemon battle form is changed.

## Why am I doing these changes?
All text should be localizable.

## What did change?
 - Change pokemon-form.ts > getSpeciesFormChangeMessage
  - Remove duplicate prefix in form change method (because preName is normally from getPokemonNameWithAffix)
  - Change hard-coded key "eternamax" -> SpeciesFormKey.ETERNAMAX
  - Change hard-coded message for localize
 - Added config, localization keys in each language folders.
 - Separated battle form message keys to distinguish between battle form(with item) and pokemon's form.
 - Translated them to Korean. (+ each language after review)

#### En
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Mega | image |
| Gigantmax | image |
| Eternamax | image |
| Other | image |
| Revert | image |

</details>

#### It
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Mega | image |
| Gigantmax | image |
| Eternamax | image |
| Other | image |
| Revert | image |

</details>

#### De
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Mega | image |
| Gigantmax | image |
| Eternamax | image |
| Other | image |
| Revert | image |

</details>

#### Fr
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Mega | image |
| Gigantmax | image |
| Eternamax | image |
| Other | image |
| Revert | image |

</details>

#### Pt_BR
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Mega | image |
| Gigantmax | image |
| Eternamax | image |
| Other | image |
| Revert | image |

</details>

#### Zh_CN
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Mega | image |
| Gigantmax | image |
| Eternamax | image |
| Other | image |
| Revert | image |

</details>

#### Zh_TW
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Mega | image |
| Gigantmax | image |
| Eternamax | image |
| Other | image |
| Revert | image |

</details>

#### Ko
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Mega | image |
| Gigantmax | image |
| Eternamax | image |
| Other | image |
| Revert | image |

</details>

## How to test the changes?
Manually (with src/overrides.ts)
Need to execute 'New Game', can test in 'check team' menu.

1. For test Mega form change message
```Typescript
  readonly STARTER_SPECIES_OVERRIDE: Species | integer = Species.MEWTWO;
  readonly STARTING_HELD_ITEMS_OVERRIDE: Array<ModifierOverride> = [ {name: "FORM_CHANGE_ITEM", type: FormChangeItem.MEWTWONITE_X}, {name: "MEGA_BRACELET"}];
```

2. For test Gigantamax
```Typescript
  readonly STARTER_SPECIES_OVERRIDE: Species | integer = Species.PIKACHU;
  readonly STARTING_HELD_ITEMS_OVERRIDE: Array<ModifierOverride> = [ {name: "FORM_CHANGE_ITEM", type: FormChangeItem.MAX_MUSHROOMS}, {name: "DYNAMAX_BAND"}];
```

3. For test Eternamax
```Typescript
  readonly STARTER_SPECIES_OVERRIDE: Species | integer = Species.ETERNATUS;
  readonly STARTING_HELD_ITEMS_OVERRIDE: Array<ModifierOverride> = [ {name: "FORM_CHANGE_ITEM", type: FormChangeItem.MAX_MUSHROOMS}, {name: "DYNAMAX_BAND"}];
```

4. For test Primal
```Typescript
  readonly STARTER_SPECIES_OVERRIDE: Species | integer = Species.KYOGRE;
  readonly STARTING_HELD_ITEMS_OVERRIDE: Array<ModifierOverride> = [ {name: "FORM_CHANGE_ITEM", type: FormChangeItem.BLUE_ORB}, {name: "MEGA_BRACELET"}];
```

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/fr/pokemon-form.ts [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
   "mega-y": "Méga-{{pokemonName}} Y",
   "primal": "Primo-{{pokemonName}}",
   "gigantamax": "{{pokemonName}} Gigamax",
   "eternamax": "{{pokemonName}} Infinimax",
 
+  "megaChange": "{{preName}} Mega Evolved\ninto {{pokemonName}}!",
+  "gigantamaxChange": "{{preName}} Gigantamaxed\ninto {{pokemonName}}!",
+  "eternamaxChange": "{{preName}} Eternamaxed\ninto {{pokemonName}}!",
+  "revertChange": "{{pokemonName}} reverted\nto its original form!",
+  "formChange": "{{preName}} changed form!",
```
Comment (Adri1) [inline src/locales/fr/pokemon-form.ts:15]:
```suggestion
  "megaChange": "{{preName}} méga-évolue\nen {{pokemonName}} !",
  "gigantamaxChange": "{{preName}} se gigamaxe\nen {{pokemonName}} !",
  "eternamaxChange": "{{preName}} devient\n{{pokemonName}} !",
  "revertChange": "{{pokemonName}} retourne\nà sa forme initiale !",
  "formChange": "{{preName}} change de forme !",
```

**Review by Adri1 — COMMENTED** [review]
French

File: src/locales/zh_CN/pokemon-form.ts [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
   "mega-y": "Mega {{pokemonName}} Y",
   "primal": "原始{{pokemonName}}",
   "gigantamax": "超极巨{{pokemonName}}",
   "eternamax": "无极巨{{pokemonName}}",
 
+  "megaChange": "{{preName}} Mega Evolved\ninto {{pokemonName}}!",
+  "gigantamaxChange": "{{preName}} Gigantamaxed\ninto {{pokemonName}}!",
+  "eternamaxChange": "{{preName}} Eternamaxed\ninto {{pokemonName}}!",
+  "revertChange": "{{pokemonName}} reverted\nto its original form!",
+  "formChange": "{{preName}} changed form!",
```
Comment (40chyan) [inline src/locales/zh_CN/pokemon-form.ts:15]:
```suggestion
  "megaChange": "{{preName}}Mega进化成了\n{{pokemonName}}！",
  "gigantamaxChange": "{{preName}}超极巨化成了\n{{pokemonName}}！",
  "eternamaxChange": "{{preName}}无极巨化成了\n{{pokemonName}}！",
  "revertChange": "{{pokemonName}}改变成了\n原本的形态！",
  "formChange": "{{preName}}改变了形态！",
```

**Review by 40chyan — COMMENTED** [review]
zhcn

File: src/locales/pt_BR/pokemon-form.ts [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
   "mega-y": "Mega {{pokemonName}} Y",
   "primal": "{{pokemonName}} Primordial",
   "gigantamax": "G-Max {{pokemonName}}",
   "eternamax": "E-Max {{pokemonName}}",
 
+  "megaChange": "{{preName}} Mega Evolved\ninto {{pokemonName}}!",
+  "gigantamaxChange": "{{preName}} Gigantamaxed\ninto {{pokemonName}}!",
+  "eternamaxChange": "{{preName}} Eternamaxed\ninto {{pokemonName}}!",
+  "revertChange": "{{pokemonName}} reverted\nto its original form!",
+  "formChange": "{{preName}} changed form!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/pokemon-form.ts:15]:
```suggestion
  "megaChange": "{{preName}} Mega Evoluiu\npara {{pokemonName}}!",
  "gigantamaxChange": "{{preName}} Gigantamaxou\npara {{pokemonName}}!",
  "eternamaxChange": "{{preName}} Eternamaxou\npara {{pokemonName}}!",
  "revertChange": "{{pokemonName}} voltou\npara sua forma original!",
  "formChange": "{{preName}} mudou de forma!",
```

File: src/locales/zh_CN/pokemon-form.ts (hunk 2 above)
Comment (mercurius-00) [inline src/locales/zh_CN/pokemon-form.ts:15]:
```suggestion
  "megaChange": "{{preName}}超级进化成了\n{{pokemonName}}！",
  "gigantamaxChange": "{{preName}}超极巨化成了\n{{pokemonName}}！",
  "eternamaxChange": "{{preName}}无极巨化成了\n{{pokemonName}}！",
  "revertChange": "{{pokemonName}}变回了\n原本的样子！",
  "formChange": "{{preName}}变成其他样子了。",
```

File: src/locales/zh_TW/pokemon-form.ts [hunk 4]
```diff
@@ -1,14 +1,21 @@
...
   "mega-y": "Mega {{pokemonName}} Y",
   "primal": "Primal {{pokemonName}}",
   "gigantamax": "G-Max {{pokemonName}}",
   "eternamax": "E-Max {{pokemonName}}",
 
+  "megaChange": "{{preName}} Mega Evolved\ninto {{pokemonName}}!",
+  "gigantamaxChange": "{{preName}} Gigantamaxed\ninto {{pokemonName}}!",
+  "eternamaxChange": "{{preName}} Eternamaxed\ninto {{pokemonName}}!",
+  "revertChange": "{{pokemonName}} reverted\nto its original form!",
+  "formChange": "{{preName}} changed form!",
```
Comment (mercurius-00) [inline src/locales/zh_TW/pokemon-form.ts:15]:
```suggestion
  "megaChange": "{{preName}}超級進化成了\n{{pokemonName}}！",
  "gigantamaxChange": "{{preName}}超極巨化成了\n{{pokemonName}}！",
  "eternamaxChange": "{{preName}}無極巨化成了\n{{pokemonName}}！",
  "revertChange": "{{pokemonName}}變回了\n原本的樣子！",
  "formChange": "{{preName}}變為其他樣子了。",
```

**Review by mercurius-00 — COMMENTED** [review]
zh-CN and zh_TW translation

File: src/locales/de/pokemon-form.ts [hunk 5]
```diff
@@ -9,6 +9,14 @@ export const pokemonForm: SimpleTranslationEntries = {
   "gigantamax": "G-Dyna-{{pokemonName}}",
   "eternamax": "U-Dyna-{{pokemonName}}",
 
+  "megaChange": "{{preName}} Mega Evolved\ninto {{pokemonName}}!",
+  "gigantamaxChange": "{{preName}} Gigantamaxed\ninto {{pokemonName}}!",
+  "eternamaxChange": "{{preName}} Eternamaxed\ninto {{pokemonName}}!",
+  "revertChange": "{{pokemonName}} reverted\nto its original form!",
+  "formChange": "{{preName}} changed form!",
+} as const;
```
Comment (CodeTappert) [inline src/locales/de/pokemon-form.ts:17]:
```suggestion
  "megaChange": "{{preName}} hat sich zu {{pokemonName}} mega-entwickelt!",
  "gigantamaxChange": "{{preName}} hat sich zu {{pokemonName}} gigadynamaximiert!",
  "eternamaxChange": "{{preName}} hat sich zu {{pokemonName}} unendynamaximiert!",
  "revertChange": "{{pokemonName}} hat seine ursprüngliche Form zurückerlangt!",
  "formChange": "{{preName}} hat seine Form geändert!",
} as const;
```

File: src/locales/it/pokemon-form.ts [hunk 6]
```diff
@@ -1,14 +1,21 @@
...
   "mega-y": "Mega {{pokemonName}} Y",
   "primal": "Archeo {{pokemonName}}",
   "gigantamax": "GigaMax {{pokemonName}}",
   "eternamax": "EternaMax {{pokemonName}}",
 
+  "megaChange": "{{preName}} Mega Evolved\ninto {{pokemonName}}!",
+  "gigantamaxChange": "{{preName}} Gigantamaxed\ninto {{pokemonName}}!",
+  "eternamaxChange": "{{preName}} Eternamaxed\ninto {{pokemonName}}!",
+  "revertChange": "{{pokemonName}} reverted\nto its original form!",
+  "formChange": "{{preName}} changed form!",
```
Comment (NicusPulcis) [inline src/locales/it/pokemon-form.ts:15]:
```suggestion
  "megaChange": "{{preName}} si evolve\nin {{pokemonName}}!",
  "gigantamaxChange": "{{preName}} si Gigamaxxizza\nin {{pokemonName}}!",
  "eternamaxChange": "{{preName}} si Dynamaxxa infinitamente\nin {{pokemonName}}!",
  "revertChange": "{{pokemonName}} è tornato\nalla sua forma originaria!",
  "formChange": "{{preName}} ha cambiato forma!",
```

**Review by NicusPulcis — APPROVED** [review]
Italian localization

**Review by Adri1 — APPROVED** [review]

**Pull request merged**